odoo/models.py

```diff
@@ -5526,17 +5526,17 @@ def filtered_domain(self, domain):
             if leaf == '|':
                 stack.append(stack.pop() | stack.pop())
             elif leaf == '!':
-                stack.append(self - stack.pop())
+                stack.append(set(self._ids) - stack.pop())
             elif leaf == '&':
                 stack.append(stack.pop() & stack.pop())
             elif leaf == expression.TRUE_LEAF:
-                stack.append(self)
+                stack.append(set(self._ids))
             elif leaf == expression.FALSE_LEAF:
-                stack.append(self.browse())
+                stack.append(set())
             else:
                 (key, comparator, value) = leaf
                 if comparator in ('child_of', 'parent_of'):
-                    stack.append(self.search([('id', 'in', self.ids), leaf], order='id'))
+                    stack.append(self._search([('id', 'in', self.ids), leaf], order='id'))
```

Comment:
It doesn't work, you compare Query with set.
`stack.append(set(self.search([('id', 'in', self.ids), leaf], order='id')._ids))`

Reply:
I was working on a branch for 13.0 just before, that's why I got confused :smile: